spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/ProcedureInput.java

```diff
@@ -68,6 +68,18 @@ public Boolean asBoolean(ProcedureParameter param, Boolean defaultValue) {
     return args.isNullAt(ordinal) ? defaultValue : (Boolean) args.getBoolean(ordinal);
   }
 
+  public Integer asInt(ProcedureParameter param) {
+    Integer value = asInt(param, null);
+    Preconditions.checkArgument(value != null, "Parameter '%s' is not set", param.name());
+    return value;
+  }
+
+  public Integer asInt(ProcedureParameter param, Integer defaultValue) {
+    validateParamType(param, DataTypes.IntegerType);
```

Comment:
Does this validation fail if there's a value which overflows int?

Reply:
yes, overflows will be parsed as `DataTypes.LongType`